probe-rs/targets/esp32c2.yaml

```diff
@@ -0,0 +1,66 @@
+name: esp32c2
```

Comment:
Lets also add the JEP106 for espressif as the manufacturer so it'll show up on the probe.rs website: https://github.com/probe-rs/probe-rs/blob/ff52ff16e2990bdcac6707a829a9a83b2604d6a4/probe-rs/targets/esp32c3.yaml#L3-L5

Reply:
I've also added it for H2 and formatted the `esp32h2.yaml` file